packages/inventory/src/services/reservation-item.ts

```diff
@@ -0,0 +1,269 @@
+    config: FindConfig<ReservationItem> = { relations: [], skip: 0, take: 10 }
+  ): Promise<[ReservationItem[], number]> {
+    const manager = this.getManager()
+    const itemRepository = manager.getRepository(ReservationItem)
+
+    const query = buildQuery(selector, config)
+    return await itemRepository.findAndCount(query)
+  }
+
+  /**
+   * Retrieves a reservation item by its id.
+   * @param reservationItemId - The id of the reservation item to retrieve.
+   * @param config - Configuration for the query.
+   * @return The reservation item with the provided id.
+   * @throws If reservationItemId is not defined or if the reservation item was not found.
+   */
+  async retrieve(
+    reservationItemId: string,
+    config: FindConfig<ReservationItem> = {}
+  ): Promise<ReservationItem> {
+    if (!isDefined(reservationItemId)) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `"reservationItemId" must be defined`
+      )
+    }
+
+    const manager = this.getManager()
+    const reservationItemRepository = manager.getRepository(ReservationItem)
+
+    const query = buildQuery({ id: reservationItemId }, config)
+    const [reservationItem] = await reservationItemRepository.find(query)
+
+    if (!reservationItem) {
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `ReservationItem with id ${reservationItemId} was not found`
+      )
+    }
+
+    return reservationItem
+  }
+
+  /**
+   * Create a new reservation item.
+   * @param data - The reservation item data.
+   * @return The created reservation item.
+   */
+  async create(data: CreateReservationItemInput): Promise<ReservationItem> {
+    const result = await this.atomicPhase_(async (manager) => {
+      const itemRepository = manager.getRepository(ReservationItem)
+
+      const inventoryItem = itemRepository.create({
+        inventory_item_id: data.inventory_item_id,
+        line_item_id: data.line_item_id,
+        location_id: data.location_id,
+        quantity: data.quantity,
+        metadata: data.metadata,
+      })
+
+      const [newInventoryItem] = await Promise.all([
+        itemRepository.save(inventoryItem),
+        this.inventoryLevelService_
+          .withTransaction(manager)
+          .adjustReservedQuantity(
+            data.inventory_item_id,
+            data.location_id,
+            data.quantity
+          ),
+      ])
+
+      return newInventoryItem
+    })
+
+    await this.eventBusService_.emit(ReservationItemService.Events.CREATED, {
+      id: result.id,
+    })
+
+    return result
+  }
+
+  /**
+   * Update a reservation item.
+   * @param reservationItemId - The reservation item's id.
+   * @param data - The reservation item data to update.
+   * @return The updated reservation item.
+   */
+  async update(
+    reservationItemId: string,
+    data: Omit<
+      DeepPartial<ReservationItem>,
+      "id" | "created_at" | "metadata" | "deleted_at"
+    >
```

Comment:
Can we add a todo here to use the update input here as well 😄 `UpdateReservationItemInput`

Reply:
Nice catch